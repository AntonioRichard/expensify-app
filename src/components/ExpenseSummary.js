import React from "react";
import { connect } from "react-redux";
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';
import numeral from "numeral";

export const ExpenseSummary = (props)=> {
    const expenseCount = props.expenses.length;
    const expenseTotal = numeral(getExpensesTotal(props.expenses) / 100).format('$0,0.00');
    return (
        <h3>
        Viewing {expenseCount === 1 ? "1 expense" : expenseCount +" expenses"} totalling {expenseTotal}
        </h3>
 );
};

const mapStateToProps = (state) => {
    return {
      expenses: selectExpenses(state.expenses, state.filters)
    };
  };

export default connect(mapStateToProps)(ExpenseSummary);