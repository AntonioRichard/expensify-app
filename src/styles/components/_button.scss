.button {
  background: $blue;
  color: #fff;
  border-style: none;
  padding: $s-size;
  font-size: large;
  display: inline-block;
  text-decoration: none;
  line-height: 1;
}

.button--link {
  background: none;
}

.button--remove {
  background: #888;
}
