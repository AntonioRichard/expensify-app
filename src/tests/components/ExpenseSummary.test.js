import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseSummary} from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test("should render text for viewing multiple expenses",()=>{
    const wrapper = shallow(<ExpenseSummary expenses={expenses}/>)
    expect(wrapper).toMatchSnapshot();
});

test("should render text for viewing single expense",()=>{
    const wrapper = shallow(<ExpenseSummary expenses={[expenses[0]]}/>)
    expect(wrapper).toMatchSnapshot();
});