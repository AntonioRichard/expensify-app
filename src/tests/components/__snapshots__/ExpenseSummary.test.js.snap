// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render text for viewing multiple expenses 1`] = `
<h3>
  Viewing 
  3 expenses
   totalling 
  $1,141.95
</h3>
`;

exports[`should render text for viewing single expense 1`] = `
<h3>
  Viewing 
  1 expense
   totalling 
  $1.95
</h3>
`;
